features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/assertj/SpanEventAssert.kt

```diff
@@ -299,6 +303,98 @@ internal class SpanEventAssert(actual: SpanEvent) :
+                        "instead was: $actualTraceId"
+                )
+                .isEqualTo(traceId)
+            val actualSpanId = actualLink.get("span_id").asString
+            val spanId = DDSpanId.toHexStringPadded(link.spanId())
+            assertThat(actualSpanId)
+                .overridingErrorMessage(
+                    "Expected SpanEvent to have span link span id: " +
+                        "$spanId but " +
+                        "instead was: $$actualSpanId"
+                )
+                .isEqualTo(spanId)
+            val traceFlags = link.traceFlags()
+            if (traceFlags.toInt() != 0) {
+                val actualTraceFlags = actualLink.get("flags").asByte
+                assertThat(actualTraceFlags)
+                    .overridingErrorMessage(
+                        "Expected SpanEvent to have span link flags: " +
+                            "$traceFlags but " +
+                            "instead was: $actualTraceFlags"
+                    )
+                    .isEqualTo(traceFlags)
+            } else {
+                assertThat(actualLink.has("flags"))
+                    .overridingErrorMessage(
+                        "Expected SpanEvent to not have span link flags but " +
+                            "instead was: $actualLink"
+                    )
+                    .isFalse()
+            }
+            val actualTraceState = actualLink.get("tracestate").asString
+            if (actualTraceState.isNotEmpty()) {
+                val traceState = link.traceState()
+                assertThat(actualTraceState)
+                    .overridingErrorMessage(
+                        "Expected SpanEvent to have span link trace state: " +
+                            "$traceState but " +
+                            "instead was: $actualTraceState"
+                    )
+                    .isEqualTo(traceState)
+            } else {
+                assertThat(actualLink.has("tracestate"))
+                    .overridingErrorMessage(
+                        "Expected SpanEvent to not have span link trace state but " +
+                            "instead was: $actualLink"
+                    )
+                    .isFalse()
+            }
+            val actualAttributes = actualLink.get("attributes").toString()
+            val expectedAttributes = link.attributes().asMap()
+            val jsonObject = JsonObject()
+            expectedAttributes.forEach { (key, value) ->
+                jsonObject.addProperty(key, value)
+            }
+            val expectedAttributesJsonString = jsonObject.toString()
+            assertThat(actualAttributes)
+                .overridingErrorMessage(
+                    "Expected SpanEvent to have span link attributes: " +
+                        "$expectedAttributesJsonString but " +
+                        "instead was: $actualAttributes"
+                )
+                .isEqualTo(expectedAttributesJsonString)
```

Comment:
not blocking: maybe extract this into a dedicated assert / split assertions? a bit hard to read